api/azimuth/provider/openstack/provider.py

```
@@ -690,6 +707,79 @@ def _tenant_network(self, create_network = False):
 
+    def _project_share(self, create_share=True):
+        self._create_manila_project_share = True
+        self._project_share_name = "johng-dev"
+
+        if not self._create_manila_project_share:
+            return
+
+        # find if project share exists
+        project_share = None
+        current_shares = self._connection.share.shares.all()
+        for share in current_shares:
+            if share.name == self._project_share_name:
+                project_share = share
+        if project_share:
+            # double check share has the correct protocol and is available
+            share_details = self._connection.share.shares.get(project_share.id)
+            self._log(f"Got share details f{share_details}")
+            if share_details.share_proto.upper() != "CEPHFS":
+                raise Exception("Currently only support CephFS shares!")
+            if share_details.status.lower() != "available":
+                raise Exception("Share is not available!")
+            if share_details.access_rules_status.lower() != "active":
+                raise Exception("Share has a problem with its access rules!")
+
+            access_list = list(self._connection.share.access.all(
+                share_id=project_share.id))
+            found_expected_access = False
+            for access in access_list:
+                if access.access_to == self._project_share_user:
+                    found_expected_access = True
+                    break
+            if not found_expected_access:
+                raise Exception("can't find the expected access rule!")
+            self._log(f"Found project share for: {self._connection.project_id}")
+
+        # no share found, create if required
+        if not project_share and create_share:
+            self._log(f"Creating project share for: {self._connection.project_id}")
+
+            # Find share type
+            default_share_type = None
```

Note Arcus's rcp-cloud-portal-demo doesn't have a default share type defined, hence https://github.com/stackhpc/ansible-slurm-appliance/pull/344 defines `cluster_home_manila_share_type`.

---

yeah, if there is only one share type, I just use that instead. We could configure it.

---

I am making it fail silently if I can't find any storage types, as on arcus it means they project isn't setup for Manila with the correct networking, etc.